### PR TITLE
account state for both jormungandr, jcli and tests

### LINKED ISSUE #491: REST account API: The delegation field format should be improved
In JSON retrieved by `GET /v0/account/get`, the `delegation` field needs a more efficient text representation for JSON, rather than a dump of a byte array.

_Originally posted by @mzabaluev in https://github.com/input-output-hk/jormungandr/issues/484#issuecomment-501695828_

### PR DESCRIPTION
**Pull request by NicolasDP**
This make sure we use the right encoding for delegation.

part of #434 
fixes #491

### CODE REVIEW

File: jormungandr-lib/src/interfaces/value.rs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+///
+/// let value: Value = StdValue(64).into();
+///
+/// println!("value: {}", value);
+///
+/// # assert_eq!(value.to_string(), "64");
+/// ```
+///
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Value(value::Value);
```
Comment (mzabaluev) [inline jormungandr-lib/src/interfaces/value.rs]: Same thing with `Value`: what's the added value?

Comment (mzabaluev) [reply jormungandr-lib/src/interfaces/value.rs]: `pub use value::Value;` is a one-line way to re-export a type from another crate.

File: jormungandr-integration-tests/src/common/jcli_wrapper/mod.rs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #![allow(dead_code)]
 
-use jormungandr_lib::interfaces::FragmentLog;
+use jormungandr_lib::interfaces::{AccountState, FragmentLog};
```
[the remark this answers is not in the document]
Comment (NicolasDP) [reply jormungandr-integration-tests/src/common/jcli_wrapper/mod.rs]: hmm, no. this is the interfaces of jomungandr. AccountState is returned from the REST API, same for FragmentLog

File: jormungandr-lib/src/crypto/key.rs [hunk 3]
```diff
@@ -80,6 +80,11 @@ impl<A: AsymmetricKey> KeyPair<A> {
 }
 
 impl<A: AsymmetricPublicKey> Identifier<A> {
+    #[inline]
+    pub fn into_public_key(self) -> PublicKey<A> {
```
[the remark this answers is not in the document]
Comment (NicolasDP) [reply jormungandr-lib/src/crypto/key.rs]: tried this, went for the simple solution of using adhoc method.

File: jormungandr-lib/src/interfaces/account_state.rs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        AccountState {
+            delegation: account.delegation().clone().map(|h| {
+                let h: [u8; 32] = h.into();
+                h.into()
+            }),
+            value: account.value().into(),
+            counter: account.get_counter(),
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (NicolasDP) [reply jormungandr-lib/src/interfaces/account_state.rs]: cannot do that. AccountState in chain_impl_mockchain does not define the same properties (especially the serde interfaces).

File: jormungandr-lib/src/interfaces/value.rs (hunk 1 above)
Comment (NicolasDP) [reply jormungandr-lib/src/interfaces/value.rs]: again, not the right interfaces/encoding/decoding

File: jormungandr-lib/src/interfaces/account_state.rs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        &self.delegation
+    }
+
+    #[inline]
+    pub fn value(&self) -> &Value {
+        &self.value
+    }
+
+    #[inline]
+    pub fn counter(&self) -> &u32 {
```
[the remark this answers is not in the document]
Comment (NicolasDP) [reply jormungandr-lib/src/interfaces/account_state.rs]: no need. but consistent with everything else.

[the remark this answers is not in the document]
Comment (mzabaluev) [reply jormungandr-lib/src/interfaces/account_state.rs]: Well, let's just fix everything else while we still can, this is embarrassing.

File: jormungandr-lib/src/interfaces/account_state.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (mzabaluev) [reply jormungandr-lib/src/interfaces/account_state.rs]: Would it be generally useful to provide Serde interfaces on mockchain? I'd say we should land these APIs there some time later and save on maintaining little special types just for jormungandr.

**Pull request merged**